### DISCUSSION TITLE
Does the HTML base plugin ignore url subfolders?

### DISCUSSION THREAD

**Discussion by kubasto** [11ty/eleventy#2743] [Q&A]
If I set such url in the _.eleventy.js_ config:
```baseHref: "http://example.com/extra/folders/"```
the result is the `/extra/folders/` part is ignored. So for example:
```<link rel="stylesheet" href="/style.css">```
becomes:
```<link rel="stylesheet" href="http://example.com/style.css">```
and not:
```<link rel="stylesheet" href="http://example.com/extra/folders/style.css">```
as expected.
Is it a bug or a feature?

**Comment by mendhak** [comment]
From my understanding, (could be wrong) it will take the baseHref value (example.com) and put that in front of whatever you return in the pathprefix at the bottom of the eleventy.js.  

 Just for comparison,  I tried this: 

```javascript
  eleventyConfig.addPlugin(EleventyHtmlBasePlugin, {
    baseHref: 'http://example.com'
  });

```

And everything was like `http://example.com/my-path-prefix/stylesheet.css`.  But when I changed baseHref to 

```
baseHref: 'http://example.com/abc/'
```

The /abc/ was ignored.  So it must be only interested in the domain part, and the pathPrefix that you return, in your case you should probably set pathPrefix="/extra/folders"?

**Reply by pdehaan** [reply]
Yeah, I think per [the docs](https://www.11ty.dev/docs/plugins/html-base/#advanced-usage), you want to set `baseHref` _and_ `pathPrefix` in this case:

```js
const { EleventyHtmlBasePlugin } = require("@11ty/eleventy");

/**
 * @param {import("@11ty/eleventy/src/UserConfig")} eleventyConfig
 * @returns {ReturnType<import("@11ty/eleventy/src/defaultConfig")>}
 */
module.exports = function (eleventyConfig) {
  eleventyConfig.addPlugin(EleventyHtmlBasePlugin, {
    baseHref: "http://example.com/",
  });

  return {
    dir: {
      input: "src",
      output: "www",
    },
    pathPrefix: "/extra/folders/",
  };
};
```

### src/index.njk
```njk
---
title: Home
layout: layouts/base.njk
---

<p>This is the home page!</p>
<a href="/about/">About</a>
```

### src/_includes/layouts/base.njk
```njk
<!Doctype HTml>
<html lang="en">
<head>
  <meta charset="utf-8"/>
  <title>{{ title }}</title>
  <link rel="stylesheet" href="/style.css" />
</head>
<body>
  <main>{{ content | safe }}</main>
</body>
</html>
```

### OUTPUT
```html
<!Doctype HTml>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Home</title>
  <link rel="stylesheet" href="http://example.com/extra/folders/style.css">
</head>
<body>
  <main>
	<p>This is the home page!</p>
	<a href="http://example.com/extra/folders/about/">About</a>
	</main>
</body>
</html>
```
